# Regressão Linear sem algorítmo

In [1]:
import pandas as pd
df  = pd.read_csv("C:/formacao_dataScience_DSA_DADOS/04_machineLearning/Cap04_Regressao_pt1/dados/pesos.csv")

In [2]:
df

Idade  Peso
0       0  3.44
1       1  4.39
2       2  4.49
3       3  4.78
4       4  5.23
5       5  6.00
6       6  6.18
7       7  7.03
8       8  7.18
9       9  7.50
10     10  8.59
11     11  8.99
12     12  9.54

Dado que estamos estimando o peso do bebê com base em sua idade, a regressão linear pode ser construída da seguinte maneira:

y = a + bx

Vamos resolver o problema a partir dos primeiros registros. Vamos supor que o conjunto de dados tenha apenas 2 pontos de dados. A modelagem ficaria assim:

Cálculo para o primeiro mês:
* y = a + b*x
* 3.44 = a + b*(0)
* a = 3.44

Cálculo para o segundo mês:
* y = a + b*x
* 4.39 = a + b*(1)
* 4.39 = 3.44 + b*(1)
* b = 4.39 - 3.44
* b = 0.95

As linhas anteriores já representam o treinamento do modelo e fomos capazes de prever os valores de a e b, nesse caso, a = 3.44 e b = 0.95. Isso é o que o modelo aprende durante o treinamento!